samples/python/lineiterator.py

```diff
@@ -0,0 +1,36 @@
+import cv2 as cv
+
+img = cv.imread(cv.samples.findFile('butterfly.jpg'))
+pt1 = (30, 40)
+pt2 = (200, 90)
+h = cv.LineIterator(pt1, pt2)
+
+pt = h.pos()
+while min(pt1[0], pt2[0]) <= pt[0] <= max(pt1[0], pt2[0]) and \
+      min(pt1[1], pt2[1]) <= pt[1] <= max(pt1[1], pt2[1]):
```

Comment:
Could we improve stop condition?

---

Could we provide a "native" Python iterator?

```
for pt in cv.LineIterator(pt1, pt2):
    ... process pt ...
```

See how to write Python (3+ only) code here: `modules/core/misc/python/package` (but for imgproc).

Python allows to replace modules's entity.
Redefine class (create new) or redefine `__iter__`/`__next__` methods:

```.py
def LineIterator__iter__(self):
    ... reset ...
    return self

def LineIterator__next__(self):
    if self.next():
      return self.pos
    else:
      raise StopIteration

cv2.LineIterator.__iter__ = LineIterator__iter__
cv2.LineIterator.__next__ = LineIterator__next__
```

Reply:
There is a problem when imort cv2 : 
    cv2.LineIterator.__iter__ = LineIterator__iter__
TypeError: cannot set '__iter__' attribute of immutable type 'cv2.LineIterator'

file __init__.py content in imgproc/python/package/lineiteraor/

```
__all__ = []

import cv2

def LineIterator__iter__(self):
    return self

def LineIterator__next__(self):
    if self.next():
        return self.pos
    else:
        raise StopIteration

cv2.LineIterator.__iter__ = LineIterator__iter__
cv2.LineIterator.__next__ = LineIterator__next__
```